### ISSUE TITLE
[Confluence] HTML contains open HTML tags.

### ISSUE THREAD

**Issue by jaredmorgs** [asciidoctor/asciidoctor#1472]
When I paste the generated Asciidoctor raw HTML into a Confluence Raw Code editor, I get a number of errors regarding the markup that I need to manually fix.

The errors are: 

`The element type "meta" must be terminated by the matching end-tag "</meta>". (400)` which relates to the open <meta> tags.

They need a closing /> or a closing </meta> for Confluence to deem them valid.

```
<head>
<meta charset="UTF-8">
<!--[if IE]><meta http-equiv="X-UA-Compatible" content="IE=edge"><![endif]-->
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Asciidoctor 1.5.2">
<title>DOC TITLE</title>
</head>
```

`The element type "img" must be terminated by the matching end-tag "</img>". (400)` which relates to the open <img> tags>.

They need a closing /> for Confluence to deem them valid.

```
<img src="https://confluence.example.com/download/attachments/54198312/LDA_Network-Logical_Overview" alt="LDA Network Logical Overview">
```

`The end-tag for element type "col" must end with a '>' delimiter. (400)` which relates to the open <col> tags>.

They need a closing /> for Confluence to deem them valid.

```
<table class="tableblock frame-all grid-all spread">
<colgroup>
<col style="width: 16%;">
<col style="width: 16%;">
<col style="width: 16%;">
<col style="width: 16%;">
<col style="width: 16%;">
<col style="width: 16%;">
</colgroup>
```

Are these issues just a Confluence vagary or are they actually not well-formed HTML that Asciidoctor is pushing out?

I'm passing the following attributes when publishing this document, in case this is part of the issue:

:experimental:
:stylesheet!:
:notitle:
:sectnums:
:images: https://confluence.example.com/download/attachments/54198335/
:toc:

**Comment by mojavelinux** [comment]
I believe this has been brought up before, but I can't remember if it has been addressed. Basically, Confluence is expecting XHTML (not HTML). Asciidoctor produces HTML by default. Fortunately, it can produce XHTML with a simple switch. That switch is on the following line:

https://github.com/asciidoctor/asciidoctor-confluence/blob/master/lib/asciidoctor/confluence.rb#L20

We need the following addition:

```ruby
@asciidoctor_options[:backend] = 'xhtml'
```

Then, Confluence will be happy.

**Comment by mojavelinux** [comment]
Technically, there is no such thing as "well-formed" HTML anymore (as of HTML5). But, the world is what it is and parsers, primarily in the Java world, try to treat HTML like XML. That's why we made the xhtml option in Asciidoctor. See https://github.com/asciidoctor/asciidoctor/issues/494

Btw, this issue should be filed in the asciidoctor-confluence repository.

...and actually, there's already an issue. Let's continue the discussion there.

https://github.com/asciidoctor/asciidoctor-confluence/issues/7

**Comment by gscheibel** [comment]
Actually what I plan to do, is to create (within asciidoctor-confluence) a
specific converter to be as close as possible to the format expected by
Confluence.
And I agree (Dan was faster than me), let's continue this conversion in the
asciidoctor-confluence issue thread.

Guillaume

**Comment by mojavelinux** [comment]
:+1:

**Comment by jaredmorgs** [comment]
Just so we're clear, this was not generated by asciidoctor-confluence.

I generated the HTML using Ruby asciidoctor, viewed source, pasted into
code source window, Confluence borked on issues (that admittedly are
probably unique to it).

I just wanted to clarify the generation method and way I got to where I am.

On Mon, Aug 24, 2015, 21:02 Dan Allen <notifications@github.com> wrote:

> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor/issues/1472#issuecomment-134151094>
> .
>
-- 

Sent from Mobile.
